[databricks-pipes] Enable usage of notebook_task for PipesDatabricksClient

SoerenStahlmann opened the pull request:
## Summary & Motivation
Currently only `spark_python_task` and `python_wheel_task` are supported for `PipesDatabricksClient`. This PR enables us to use `notebook_task` as well.

## Changelog
* Add support for `notebook_task` in `PipesDatabricksClient`

maximearmstrong reviewed (changes requested):
I believe the code is correct, but we would need some form of testing. 

When using base parameters, you'll need to use `PipesDatabricksNotebookWidgetsParamsLoader` when opening the pipe, like so:

```python
from dagster_pipes import (
    PipesDatabricksNotebookWidgetsParamsLoader,
    open_dagster_pipes,
)

with open_dagster_pipes(
    context_loader=...,
    message_writer=...,
    params_loader=PipesDatabricksNotebookWidgetsParamsLoader(dbutils.widgets),
) as pipes:
    # Do something with your pipe
```

Can you confirm that this have been successfully tested? 

Adding a test suite in `dagster-databricks/dagster_databricks_test/test_pipes.py` would also be required - or you could use an adapt an existing one.

SoerenStahlmann commented:
Hey @maximearmstrong ,

I can confirm I tested the changes successfully manually and now also added the appropriate test case.

The example section was also added.

If the code needs any refinement just let me know.
